### ISSUE TITLE
Unchecked null in fetchAuthToken

### ISSUE THREAD

**Issue by fulldecent** [googleapis/google-auth-library-php#398]
This line of code https://github.com/googleapis/google-auth-library-php/blob/3a1a5c5b5a3006b3d05256d5df7066f37252112c/src/Middleware/AuthTokenMiddleware.php#L122

is able to return null. But it is asserted non-null in the next line of code.

https://github.com/googleapis/google-auth-library-php/blob/3a1a5c5b5a3006b3d05256d5df7066f37252112c/src/Middleware/AuthTokenMiddleware.php#L124

That creates run time errors that are hard to diagnose.

> .../vendor/google/auth/src/Middleware/AuthTokenMiddleware.php:124
>
> array_key_exists() expects parameter 2 to be array, null given

# Recommended fix

- [ ] Adopt [type declarations](https://www.php.net/manual/en/language.types.declarations.php) here and in all Google's PHP API libraries. Static testing will find these kinds of errors.

#### Environment details

  - OS: CentOS 7
  - PHP version: 7.4
  - Package name and version: apiclient / ^2

**Comment by bshaffer** [comment]
We use phpstan for static analysis and are currently on [level 7](https://github.com/googleapis/google-auth-library-php/blob/main/phpstan.neon.dist#L4). This is definitely something it should catch.

**Comment by bshaffer** [comment]
Running through the fetchers we have,

 - [GCECredentials](https://github.com/googleapis/google-auth-library-php/blob/main/src/Credentials/GCECredentials.php#L366) - does not return null
 - [ServiceAccountJwtAccessToken](https://github.com/googleapis/google-auth-library-php/blob/main/src/Credentials/ServiceAccountJwtAccessCredentials.php#L142) - returns `null`
 - [OAuth2](https://github.com/googleapis/google-auth-library-php/blob/main/src/OAuth2.php#L538) - does not return `null`
 - All other fetchers are either deprecated or return the result of `OAuth2::fetchAuthToken`

So we can just update `ServiceAccoutnJwtAccessCredentials` to return an empty array instead of `null`, as this goes against the `FetchAuthTokenInterface`. As this technically breaks backwards compatibility, alternately we could just cast the result of the call to an array in `AuthTokenMiddleware`

**Comment by poef** [comment]
Running the same code with latest main branche now gives a new, also cryptic error message

```
PHP Warning:  fread(): SSL operation failed with code 1. OpenSSL Error messages:
error:0A000126:SSL routines::unexpected eof while reading in .../google-test/vendor/guzzlehttp/psr7/src/Stream.php on line 232
```

**Comment by bshaffer** [comment]
This fix has been released in [v1.21.1](https://github.com/googleapis/google-auth-library-php/releases/tag/v1.21.1)
